armory/data/datasets.py

```diff
@@ -778,10 +778,18 @@ def canonical_audio_preprocess(context, batch):
             assert x.min() >= context.input_min
             assert x.max() <= context.input_max
 
-        batch = np.array(
-            [x.astype(context.output_type) / context.quantization for x in batch],
-            dtype=object,
-        )
+        items_are_different_lengths = np.any([len(i)-len(batch[0]) for i in batch])
+
+        if items_are_different_lengths or len(batch) == 1:
```

Comment:
When `len(batch) == 1`, this will fail the later assertion of `dtype`, as the input type is `np.object`. I would remove the `or len(batch) == 1` portion.

If a batch of size 1 is sent in that is not an object array, it won't hit this code.